lang/cs/Org.Apache.REEF.Evaluator.Tests/ContextRuntimeTests.cs

```diff
@@ -149,6 +155,176 @@ public void TestBaseServiceWithContextStacking()
+                    Assert.False(ReferenceEquals(
+                        contextRuntime.ContextInjector.GetInstance<TestContextEventHandler>(),
+                        childContextRuntime.ContextInjector.GetInstance<TestContextEventHandler>()));
+                }
+            }
+        }
+
+        [Fact]
+        [Trait("Priority", "0")]
+        [Trait("Category", "Unit")]
+        public void TestContextStackingParentContext()
+        {
+            var serviceInjector = TangFactory.GetTang().NewInjector();
+            var contextConfig = GetSimpleContextConfiguration();
+
+            using (var contextRuntime = new ContextRuntime(serviceInjector, contextConfig, Optional<ContextRuntime>.Empty()))
+            {
+                var childContextConfiguration = GetSimpleContextConfiguration();
+                using (var childContextRuntime = contextRuntime.SpawnChildContext(childContextConfiguration))
+                {
+                    Assert.False(contextRuntime.ParentContext.IsPresent());
+                    Assert.True(childContextRuntime.ParentContext.IsPresent());
+                    Assert.True(ReferenceEquals(contextRuntime, childContextRuntime.ParentContext.Value));
+                }
+            }
+        }
+
+        [Fact]
+        [Trait("Priority", "0")]
+        [Trait("Category", "Unit")]
+        public void TestUnableToSpawnChildWhileTaskIsRunning()
+        {
+            var serviceInjector = TangFactory.GetTang().NewInjector();
+            var contextConfig = GetSimpleContextConfiguration();
+
+            using (var contextRuntime = new ContextRuntime(serviceInjector, contextConfig, Optional<ContextRuntime>.Empty()))
+            {
+                var taskConfig = TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Task, GenericType<TestTask>.Class)
+                    .Set(TaskConfiguration.Identifier, "ID")
+                    .Build();
+
+                try
+                {
+                    var hbMgr = Substitute.For<IHeartBeatManager>();
+                    contextRuntime.ContextInjector.BindVolatileInstance(GenericType<IHeartBeatManager>.Class, hbMgr);
+                    contextRuntime.StartTask(taskConfig, hbMgr);
+
+                    Assert.True(contextRuntime.TaskRuntime.IsPresent());
+                    Assert.True(contextRuntime.GetTaskStatus().IsPresent());
+                    Assert.Equal(contextRuntime.GetTaskStatus().Value.state, State.RUNNING);
+
+                    var childContextConfiguration = GetSimpleContextConfiguration();
+
+                    Assert.Throws<InvalidOperationException>(
+                        () => contextRuntime.SpawnChildContext(childContextConfiguration));
+                }
+                finally
+                {
+                    var testTask = contextRuntime.TaskRuntime.Value.Task as TestTask;
+                    if (testTask == null)
+                    {
+                        throw new Exception();
```

Comment:
Why do we throw Exception in tests? If we need to fail the test, a respective Assert seems to be a proper way to do so.